openreview/venue/venue.py

```diff
@@ -1088,10 +1088,21 @@ def ithenticate_create_and_upload_submission(self):
                 owner = (
                     submission.signatures[0]
                     if submission.signatures[0].startswith("~")
-                    else submission.content["authorids"]["value"][0]
+                    else self.client.get_note_edits(note_id=submission.id, invitation=self.get_submission_id(), sort='tcdate:asc')[0].signatures[0]
                 )
+                print(f"Creating submission for {submission.id} with owner {owner}")
                 owner_profile = self.client.get_profile(owner)
 
+                eula_version = submission.content.get("iThenticate_agreement", {}).get("value", "v1beta").split(":")[-1].strip()
+                
+                iThenticate_client.accept_EULA(
```

Comment:
is this a new request required?

Reply:
you are passing the profile.id as user_id but I don't see the user_id in the create_submission call, how iThenticate matches the agreement with the submission?

Reply:
This is so that iThenticate has it in their record that the user accepted EULA, what version they accepted, and the timestamp.

Reply:
> you are passing the profile.id as user_id but I don't see the user_id in the create_submission call, how iThenticate matches the agreement with the submission?

I believe iThenticate uses the owner field from the create_submission call as the user_id

Reply:
the owner field might not be the same than the user_id.  An author can sign the edit using a tilde id but their profile.id can be different. 

In order to make sure they match, you need to pass `owner=owner_profile.id`

Reply:
Made the change. Thanks for catching this.